File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.swift
import Foundation

func solution(_ s:String) -> Int {
    var answer : String = "";
    var s_list = s.map{ String($0) }
    var word = ""
    
    for c in s_list {
        let v = Int(UnicodeScalar(c)!.value)
        if (48...57) ~= v {  answer += "\(c)" }
        else { 
            word += c 
            if (word.length > 0) {
                let converted = convert(word)
                if (converted != -1) {
                    answer += "\(convert(word))"
                    word = ""
                }
            }
        }
    }
    if (word.length > 0) {
        let converted = convert(word)
        if (converted != -1) {
            answer += "\(convert(word))"
            word = ""
        }
    }
    
    return Int(answer)!
}

func convert(_ word: String) -> Int {
    let num_of_word = [
        "zero":0,
        "one":1,
        "two":2,
        "three":3,
        "four":4,
        "five":5,
        "six":6,
        "seven":7,
        "eight":8,
        "nine":9,
    ]
    for key in num_of_word.keys {
        if (word == key) { return num_of_word[key]! }
    }
    return -1
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {
        String answer = "";
        
        String temp = "";
        for(int i=0; i<s.length(); i++) {
            String converted = convertToNum(temp);
            if (!converted.isEmpty()) {
                answer += converted;
                temp = "";
            }
            if (Character.isDigit(s.charAt(i))) {
                answer += Character.toString(s.charAt(i));
            } else {
                temp += Character.toString(s.charAt(i));
            }
        }
        if(!temp.isEmpty()) answer += convertToNum(temp);
        
        return Integer.parseInt(answer);
    }
    
    public String convertToNum(String num_str) {
        switch(num_str) {
            case "zero": return "0";
            case "one": return "1";
            case "two": return "2";
            case "three": return "3";
            case "four": return "4";
            case "five": return "5";
            case "six": return "6";
            case "seven": return "7";
            case "eight": return "8";
            case "nine": return "9";
        }
        return "";
    }
}